### PR TITLE
Style Selection: Add theme style variation icons

### PR DESCRIPTION
**Pull request by taipeicoder**
#### Proposed Changes

This PR adds available style variations to theme meta. See design spec: 7l53h5fxAUNjVRBQ82YFwn-fi-238%3A41198
Clicking any variation icon should open the full-screen theme preview, although it's worth mentioning that the entire section is clickable since it's within a button element.

Screen Shot 2022-08-24 at 5 22 21 PM

Note that this PR depends on D86264-code, and because of that the API payload might change.

#### Testing Instructions

* Head to the unified design picker `/setup/designSetup?siteSlug=${site_slug}`.
* Ensure that the style variation icons are as per design spec.
* Ensure that clicking on any style variation icon should open the full-screen theme preview.

#### Pre-merge Checklist

- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?

Related to #66734

### CODE REVIEW

File: packages/data-stores/src/starter-designs-queries/use-starter-designs-query.ts [hunk 1]
```diff
@@ -26,12 +31,14 @@ interface StaticDesign {
...
 	categories: Category[];
 	price?: string;
+	style_variations?: ThemeStyleVariation[];
 }
 
 interface GeneratedDesign {
 	slug: string;
 	title: string;
 	recipe: DesignRecipe;
+	style_variations?: ThemeStyleVariation[];
```
Comment (fushar) [inline packages/data-stores/src/starter-designs-queries/use-starter-designs-query.ts]: We're not returning `style_variations` for generated designs (at least for now) 😄

File: packages/design-picker/src/components/theme-style-variations-buttons/theme-style-variations-button.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import { createElement, useMemo } from 'react';
+import type { ThemeStyleVariation, ThemeStyleVariationSettingsColorPalette } from '../../types';
+import './style.scss';
+
+interface ThemeStyleVariationsButtonProps {
+	tagName?: string;
+	variation?: ThemeStyleVariation;
+}
+
+const ThemeStyleVariationsButton: React.FC< ThemeStyleVariationsButtonProps > = ( {
```
Comment (fushar) [inline packages/design-picker/src/components/theme-style-variations-buttons/theme-style-variations-button.tsx]: Hmm, technically it is not a "button". What about `ThemeStyleVariationBadge` instead?

File: packages/design-picker/src/types.ts [hunk 3]
```diff
@@ -21,6 +21,24 @@ export interface DesignRecipe {
 	footer_pattern_ids?: number[];
 }
 
+export interface ThemeStyleVariation {
+	slug: string;
+	title: string;
```
Comment (fushar) [inline packages/design-picker/src/types.ts]: Now the backend doesn't return `title` as it is not used anywhere.

**Review by fushar — COMMENTED** [review]
Looking great 🔥 added some comments, otherwise it is beautiful 😄

File: packages/data-stores/src/starter-designs-queries/use-starter-designs-query.ts (hunk 1 above)
Comment (taipeicoder) [reply packages/data-stores/src/starter-designs-queries/use-starter-designs-query.ts]: Good catch, removed!

File: packages/design-picker/src/components/theme-style-variations-buttons/theme-style-variations-button.tsx (hunk 2 above)
Comment (taipeicoder) [reply packages/design-picker/src/components/theme-style-variations-buttons/theme-style-variations-button.tsx]: Makes sense 👍

File: packages/design-picker/src/components/theme-style-variations-badges/theme-style-variations-badge.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import { createElement, useMemo } from 'react';
+import type { ThemeStyleVariation, ThemeStyleVariationSettingsColorPalette } from '../../types';
+import './style.scss';
+
+interface ThemeStyleVariationsBadgeProps {
+	tagName?: string;
+	variation?: ThemeStyleVariation;
+}
+
+const ThemeStyleVariationsBadge: React.FC< ThemeStyleVariationsBadgeProps > = ( {
```
Comment (fushar) [inline packages/design-picker/src/components/theme-style-variations-badges/theme-style-variations-badge.tsx]: [super nit] [grammar] `ThemeStyleVariationBadge` (+ the props, and filename) 🙏 (as this is for 1 variation only)

File: packages/design-picker/src/components/theme-style-variations-badges/theme-style-variations-badges.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+import type { ThemeStyleVariation } from '../../types';
+import './style.scss';
+
+interface ThemeStyleVariationsBadgesProps {
+	tagName?: string;
+	maxVariationsToShow?: number;
+	variations: ThemeStyleVariation[];
+}
+
+const ThemeStyleVariationsBadges: React.FC< ThemeStyleVariationsBadgesProps > = ( {
```
Comment (fushar) [inline packages/design-picker/src/components/theme-style-variations-badges/theme-style-variations-badges.tsx]: [super nit] [grammar] `ThemeStyleVariationBadges` (+ the props, and filename) 🙏

File: packages/design-picker/src/components/theme-style-variations-badges/theme-style-variations-badge.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+	tagName = 'button',
```
Comment (fushar) [inline packages/design-picker/src/components/theme-style-variations-badges/theme-style-variations-badge.tsx]: We seem to always use `<span/>`. Why do we need to make this a prop? Could we just hardcode the `<span/>` ?

Comment (taipeicoder) [reply packages/design-picker/src/components/theme-style-variations-badges/theme-style-variations-badge.tsx]: Originally, I planned it to be a `<button>`, but we can't have `<button>` nested in a `<button>` 🥲
Removing `tagName` as we don't have the need for it now.

**Review by fushar — APPROVED** [review]
🚢

**Pull request merged**